mmocr/datasets/pipelines/spotting_transforms.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from mmdet.core import PolygonMasks
+from mmdet.datasets.builder import PIPELINES
+from scipy.special import comb as n_over_k
+
+from mmocr.utils import check_argument
+
+
+@PIPELINES.register_module()
+class GenerateBezierPoints:
+    """Generate bezier curve annotation for ABCNet: Real-time Scene Text
+    Spotting with Adaptive Bezier-Curve Network.
+
+    [https://arxiv.org/abs/2002.10200].
+    """
+
+    def __init__(self):
+        ...
+
+    def bezier_coeff(self, ts):
+
+        def mtk(n, t, k):
+            return t**k * (1 - t)**(n - k) * n_over_k(n, k)
+
+        return [[mtk(3, t, k) for k in range(4)] for t in ts]
+
+    def bezier_fit(self, xs, ys):
+        dy = ys[1:] - ys[:-1]
+        dx = xs[1:] - xs[:-1]
+        dt = (dx**2 + dy**2)**0.5
+        t = dt / dt.sum()
+        t = np.hstack(([0], t))
+        t = t.cumsum()
+
+        data = np.column_stack((xs, ys))
+        pseudo_inv = np.linalg.pinv(self.bezier_coeff(t))
+
+        control_points = pseudo_inv.dot(data)
+        return control_points[1:-1]
+
+    def __call__(self, results):
+        polygon_masks = results['gt_masks'].masks
+        assert check_argument.is_2dlist(polygon_masks)
+
+        bezier_points = []
+        for poly in polygon_masks:
+            poly = poly[0]
+            assert len(poly) % 4 == 0
+            num_points = int(len(poly) / 2)
+            num_side_points = int(len(poly) / 4)
```

Comment:
Two assumptions here: 1. there must be 2k points in the polygonmask; 2. the first and last k points forms side curves. I'm concerned about its generalizability: Is this transform applicable to *all* kinds of point annotations? If not, shall we provide util to format the points in this way?